app/src/main/res/layout/fragment_plant_detail.xml

```diff
@@ -166,13 +166,12 @@
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:layout_margin="@dimen/fab_margin"
-            android:onClick="@{() -> callback.add(viewModel.plant)}"
+            android:onClick="@{() -> viewModel.isPlanted ?  callback.remove(viewModel.plant) : callback.add(viewModel.plant)}"
```

Comment:
I wonder if it wouldn't be better to add this logic to the ViewModel itself, rather than having it here in the binding.

Reply:
That would be better.